packages/expo-cli/src/commands/build/AndroidBuilder.ts

```diff
@@ -118,58 +118,50 @@ See https://docs.expo.io/distribution/building-standalone-apps/#2-configure-appj
       (this.options.clearCredentials || !credentialsExist)
     ) {
       console.log('');
-      const questions: Question[] = [
+      const questions: PromptObject[] = [
         {
-          type: 'rawlist',
+          type: 'select',
           name: 'uploadKeystore',
           message: `Would you like to upload a keystore or have us generate one for you?\nIf you don't know what this means, let us handle it! :)\n`,
           choices: [
-            { name: 'Let Expo handle the process!', value: false },
-            { name: 'I want to upload my own keystore!', value: true },
+            { title: 'Let Expo handle the process!', value: false },
+            { title: 'I want to upload my own keystore!', value: true },
           ],
         },
         {
-          type: 'input',
+          type: (prev, values) => values.uploadKeystore && 'text',
           name: 'keystorePath',
           message: `Path to keystore:`,
-          validate: async (keystorePath: string): Promise<boolean> => {
+          validate: async (keystorePath: string): Promise<boolean | string> => {
             try {
               const keystorePathStats = await fs.stat(keystorePath);
               return keystorePathStats.isFile();
             } catch (e) {
-              // file does not exist
-              console.log('\nFile does not exist.');
-              return false;
+              return 'File does not exist.';
```

Comment:
- Returning `true` passes validation
- Returning `false` shows the default error message
- Returning any `string` shows the string as the error message